xwiki/templates/deployment.yaml

```diff
@@ -0,0 +1,27 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
```

Comment:
Since XWiki is stateful - this should be a stateful set. You should also add a Persistent volume.